step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/LayeredAccessor.java

```diff
@@ -148,6 +149,21 @@ public void save(Iterable<T> entities) {
 		getAccessorForPersistence().save(entities);
 	}
 
+	@Override
+	public Stream<VersionableEntity> getHistory(ObjectId id, Integer skip, Integer limit) {
+		return layeredLookup(a->a.getHistory(id, skip, limit));

```

Comment:
Not sure if this would work in case of one of the accessors doesn't support history.

Reply:
it would break, I changed the impementation to only call it if the accessor has versioning enabled, same for restore.